src/lava/proc/io/bridge/extractor.py

```diff
@@ -0,0 +1,179 @@
+        (1) When the channel buffer is empty, receiving will either:
+            (a) Block until an item is available (until the sending part
+            sends an item).
+            (b) Not block, and the received data will be zeros.
+        (2) When the channel buffer is not empty, receiving will either:
+            (a) Receive a single item, being the oldest item put in the
+            channel (FIFO).
+            (b) Receive all items available, accumulated.
+
+    Parameters
+    ----------
+    shape : tuple
+        Shape of the InPort of the Process.
+    buffer_size : optional, int
+        Buffer size of the extractor_channel.
+    extractor_channel_config : optional, ChannelConfig
+        Configuration object specifying how the src_port behaves when the
+        buffer is full and how the dst_port behaves when the buffer is empty
+        and not empty.
+    """
+    def __init__(self,
+                 shape: tuple[int, ...],
+                 buffer_size: ty.Optional[int] = 50,
+                 extractor_channel_config: ty.Optional[ChannelConfig] =
+                 ChannelConfig()) -> None:
+        super().__init__()
+
+        validate_shape(shape)
+        validate_buffer_size(buffer_size)
+        validate_channel_config(extractor_channel_config)
+
+        self._shape = shape
+
+        self._extractor_channel_config = extractor_channel_config
+
+        self._multi_processing = MultiProcessing()
+        self._multi_processing.start()
+
+        self._extractor_channel = \
+            PyPyChannel(message_infrastructure=self._multi_processing,
+                        src_name="src",
+                        dst_name="dst",
+                        shape=self._shape,
+                        dtype=float,
+                        size=buffer_size)
+        self._extractor_channel_dst_port = self._extractor_channel.dst_port
+        self._extractor_channel_dst_port.start()
+
+        self.proc_params["shape"] = shape
+        self.proc_params["extractor_channel_config"] = extractor_channel_config
+        self.proc_params["extractor_channel_src_port"] = \
+            self._extractor_channel.src_port
+
+        if self._extractor_channel_config.recv_buffer_empty == \
+                ChannelRecvBufferEmpty.BLOCKING:
+            self._recv_empty = recv_empty_blocking
+        elif self._extractor_channel_config.recv_buffer_empty == \
+                ChannelRecvBufferEmpty.NON_BLOCKING_ZEROS:
+            self._recv_empty = recv_empty_non_blocking_zeros
+
+        if self._extractor_channel_config.recv_buffer_not_empty == \
+                ChannelRecvBufferNotEmpty.FIFO:
+            self._recv_not_empty = recv_not_empty_fifo
+        elif self._extractor_channel_config.recv_buffer_not_empty == \
+                ChannelRecvBufferNotEmpty.ACCUMULATE:
+            self._recv_not_empty = recv_not_empty_accumulate
+
+        self.in_port = InPort(shape=shape)
+
+    def recv_data(self) -> np.ndarray:
```

Comment:
I'm not a fan of gratuitous name shortening. Good, public APIs should prioritize clarity first, then conciseness.

Data is redundant, since everything that is ever sent or received in a computer is data, so a better name would be just receive.

Reply:
Changed this to receive, and send_data in the Injector to send.